### DISCUSSION TITLE
Print Showing Tooltip into Canvas

### DISCUSSION THREAD

**Discussion by pandika-csgi** [clientIO/joint#1726] [Q&A]
Hi, it's me again 

Is there an approach to print paper include the tooltip?
So, I'm already showing a tooltip on paper, and if I click print, the tooltip not include
any solution/setting to include the tooltip?

Thanks

**Comment by kumilingus** [comment]
It's not going to work well. The position of the tooltips depend the current paper transformation and the paper print export might have a different one. You could further enhance the printed paper with [`ready()` callback](https://resources.jointjs.com/docs/rappid/v3.5/format.Print.html#format.Print) and display the tooltips again to fix their positions. And make the tooltips visible (they are hidden by default by `format.Print` plugin). I don't believe this is the right approach though.

Perhaps, the next step will be to export the paper to SVG / PNG. Well, in that case there will be no way to display the HTML tooltips in the export.

A better approach was to create tooltips as JointJS elements and add them to the graph and then export. Any SVG that is part of the paper will be exported out-of-the-box (e.g `linkTools` / `elementTools` could be exported too, but not `Halo` nor `Selection` as they are built of HTML and rendered above the paper).

Alternatively a tooltip could be defined  as a [custom highlighter](https://resources.jointjs.com/docs/jointjs/v3.5/joint.html#highlighters.custom) class (again an SVG).

**Reply by pandika-csgi** [reply]
Thanks again @kumilingus, better approach was to create tooltips as JointJS elements and add them to graph, can you guide me to this approach like example code or documentation about it?

**Reply by kumilingus** [reply]
Create a custom JointJS element that looks like a tooltip.

- tutorial: https://resources.jointjs.com/tutorial/custom-elements
- example: https://jsfiddle.net/kumilingus/97dwxqv3/
